### PR TITLE
added password hashing

### PR DESCRIPTION
**Pull request by NimishaGarg**

### CODE REVIEW

File: src/main/java/controller/UserAPI.java [hunk 1]
```diff
@@ -100,7 +100,8 @@
         requestValidator.loginRequestValidator(data.keySet(), path);
         String username = data.get("Username").toString();
         String password = data.get("Password").toString();
-        long id = userAccessor.login(username, password);
+        String passwordHash = User.hashPassword(password);
```
Comment (sankalpbansal-google) [inline src/main/java/controller/UserAPI.java:103]: "passwordHash" is not serving any purpose.
Lets write: .login(username, User.hashPassword(password))

File: src/main/java/controller/UserAPI.java [hunk 2]
```diff
@@ -100,7 +100,8 @@
...
+        long id = userAccessor.login(username, passwordHash);
         if(id == -1) {
```
Comment (sankalpbansal-google) [inline src/main/java/controller/UserAPI.java:105]: Did we make sure, that the random generated id can never be negative (or -1 to be specific)?

Comment (NimishaGarg) [reply src/main/java/controller/UserAPI.java:105]: Yes the range of random numbers being generated is from 1 to `Long.MAX_VALUE`.

File: src/main/java/controller/UserAPI.java (hunk 1 above)
Comment (NimishaGarg) [reply src/main/java/controller/UserAPI.java]: Done.

**Pull request merged**